zap/src/main/java/org/parosproxy/paros/extension/history/ExtensionHistory.java

```diff
@@ -269,6 +275,7 @@ public void hook(ExtensionHook extensionHook) {
             extensionHook.getHookMenu().addPopupMenuItem(getPopupMenuJumpTo());
             // ZAP: Added history notes
             extensionHook.getHookMenu().addPopupMenuItem(getPopupMenuNote());
+            extensionHook.getHookMenu().addPopupMenuItem(getPopupMenuSiteNote());
```

Comment:
`new PopupMenuSiteNote(getView()))`

Reply:
> `new PopupMenuSiteNote(getView()))`

why we would pass getView to PopupMenuSiteNote it need extensionHistory as parameter ? i have take the reference from PopMenuNote.

Reply:
Because it's the only dependency it has, the dialog can live in the pop up menu item (less API exposure overall, this should be moved to an add-on at some point).

Reply:
sorry but still didn't get it.

Reply:
I think he is suggesting the only reason it passes the extension is for its view.

Reply:
Move the dialogue from the the extension to the pop up menu class, and it only needs to pass the view.

Reply:
done

Reply:
The getter is not necessary.

Reply:
are you suggesting to remove this line ? then how will PopupMenuSiteNote will get instantiate ? 
`extensionHook.getHookMenu().addPopupMenuItem(getPopupMenuSiteNote());`

Reply:
Instantiate it directly, as mentioned in the first comment.

Reply:
done

Reply:
The unused `popupMenuSiteNote` instance variable can be removed.

Reply:
do you mean the comments in ExtensionHistory file ? i think the previous comment is still appropriate.
`// ZAP: 2024/01/25 Added SiteNotesAddDialog and its utility methods (Issue 8240).`

Reply:
It's just adding the popup menu item now.